core/templates/components/entity-creation-services/exploration-creation.service.spec.ts

```diff
@@ -0,0 +1,224 @@
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+/**
+ * @fileoverview Unit test for Exploration creation service.
+ */
+
+import { HttpClientTestingModule } from '@angular/common/http/testing';
+import { fakeAsync, TestBed, tick } from '@angular/core/testing';
+import { NgbModal, NgbModalRef } from '@ng-bootstrap/ng-bootstrap';
+import { UrlInterpolationService } from 'domain/utilities/url-interpolation.service';
+import { AlertsService } from 'services/alerts.service';
+import { WindowRef } from 'services/contextual/window-ref.service';
+import { CsrfTokenService } from 'services/csrf-token.service';
+import { LoaderService } from 'services/loader.service';
+import { SiteAnalyticsService } from 'services/site-analytics.service';
+import { ExplorationCreationBackendApiService, ExplorationCreationResponse } from './exploration-creation-backend-api.service';
+import { ExplorationCreationService } from './exploration-creation.service';
+
+class MockExploratinoCreationBackendApiService {
+  throwError: boolean = false;
+  message: string;
+  registerNewExplorationAsync(): object {
+    return {
+      then: (
+          successCallback: (response: {explorationId: string}) => void,
+          errorCallback: (errorMessage: string) => void
+      ) => {
+        if (this.throwError) {
+          errorCallback(this.message);
+        } else {
+          successCallback({
+            explorationId: 'exp1'
+          });
+        }
+      }
+    };
+  }
+}
+
+class MockWindowRef {
+  _window = {
+    location: {
+      _href: '',
+      get href(): string {
+        return this._href;
+      },
+      set href(val) {
+        this._href = val;
+      }
+    }
+  };
+  get nativeWindow() {
+    return this._window;
+  }
+}
+
+describe('ExplorationCreationService', () => {
+  let ecs: ExplorationCreationService;
+  let ecbas: MockExploratinoCreationBackendApiService;
+  let loaderService: LoaderService;
+  let siteAnalyticsService: SiteAnalyticsService;
+  let urlInterpolationService: UrlInterpolationService;
+  let csrfTokenService: CsrfTokenService;
+  let alertsService: AlertsService;
+  let windowRef: MockWindowRef;
+  let ngbModal: NgbModal;
+
+  beforeEach(() => {
+    windowRef = new MockWindowRef();
+    TestBed.configureTestingModule({
+      imports: [HttpClientTestingModule],
+      providers: [
+        {
+          provide: ExplorationCreationBackendApiService,
+          useClass: MockExploratinoCreationBackendApiService
+        },
+        {
+          provide: WindowRef,
+          useValue: windowRef
+        }
+      ]
+    });
+
+    ecs = TestBed.inject(ExplorationCreationService);
+    ecbas = (TestBed.inject(
+      ExplorationCreationBackendApiService) as unknown) as
```

Comment:
Not using `as unknown` gives this error:

```

Reply:
Not using `as unknown` gives this error:

`
Conversion of type 'ExplorationCreationBackendApiService' to type 'MockExploratinoCreationBackendApiService' may be a mistake because neither type sufficiently overlaps with the other. If this was intentional, convert the expression to 'unknown' first.
`

Reply:
Are we still using the MockService?

Reply:
Nope, removed the mock